src/api/java/mekanism/api/chemical/Chemical.java

```diff
@@ -1,120 +1,251 @@
+                // when the radiation manager is disabled
+                this.hasAttributesWithValidation = this.hasLegacyAttributesWithValidation = true;
+            }
+        }
+        this.oreTag = this.legacyOreTag = builder.getOreTag();
         this.isGaseous = builder.isGaseous();
     }
 
     @Override
-    public String toString() {
-        return "[Chemical: " + getRegistryName() + "]";
+    public final String toString() {
+        return MekanismAPI.CHEMICAL_REGISTRY.wrapAsHolder(this).getRegisteredName();
     }
 
     @NotNull
     @Override
-    public Chemical getChemical() {
+    public final Chemical getChemical() {
         return this;
     }
 
     @Override
     public String getTranslationKey() {
         if (translationKey == null) {
-            translationKey = Util.makeDescriptionId("chemical", getRegistryName());
+            //Note: Because chemical registry has a default key, we have to query the name via the resource key so that we can pass null
+            // to makeDescriptionId in cases when our chemical is unregistered
+            translationKey = Util.makeDescriptionId("chemical", MekanismAPI.CHEMICAL_REGISTRY.getResourceKey(this)
+                  .map(ResourceKey::location)
+                  .orElse(null)
+            );
         }
         return translationKey;
     }
 
+    @SuppressWarnings("removal")
+    @Deprecated(forRemoval = true, since = "10.7.11")
+    private Map<Class<? extends ChemicalAttribute>, ChemicalAttribute> getAllAttributes() {
+        if (attributes.isEmpty()) {//If we have not attributes attached, just return the legacy attributes
+            return legacyAttributeMap;
+        }
+        if (attributeMap == null) {
+            attributeMap = new HashMap<>(legacyAttributeMap);
+            for (IChemicalAttribute attribute : attributes) {
+                ChemicalAttribute legacyAttribute = attribute.toLegacyAttribute();
+                attributeMap.put(legacyAttribute.getClass(), legacyAttribute);
+            }
+        }
+        return attributeMap;
+    }
+
     @Override
+    @SuppressWarnings("removal")
+    @Deprecated(forRemoval = true, since = "10.7.11")
     public boolean has(Class<? extends ChemicalAttribute> type) {
-        return attributeMap.containsKey(type);
+        return getAllAttributes().containsKey(type);
+    }
+
+    /**
+     * Whether this chemical has a legacy attribute of a certain type. This method explicitly only checks legacy (in code) defined attributes, and will return false in
+     * cases where there is only a modern attribute declared via a data map.
+     *
+     * @param type The type of the attribute to check for.
+     *
+     * @return if this chemical has the attribute.
+     *
+     * @since 10.7.11
+     */
+    @SuppressWarnings("removal")
+    @Deprecated(forRemoval = true, since = "10.7.11")
+    public boolean hasLegacy(Class<? extends ChemicalAttribute> type) {
```

Comment:
the uses of this look like they could be helpers added to this class

Reply:
Personally I am a bit against that, mostly since we use each only in one place, and I wouldn't want to be just calling the helpers directly as I don't think the helpers make sense to keep after the legacy way is gone.